### PR TITLE
CVE-2017-15707

### PR DESCRIPTION
**Pull request by Dmj1097**
This is the update document with information relevant to all aspects of the outdated json-lib library allowing for DOS attack vulnerability.

### CODE REVIEW

File: cves/CVE-2017-15707.yml [hunk 1]
```diff
@@ -55,15 +60,15 @@ bounty:
...
   announced:
   url:
-bugs: []
+bugs: [WW-4892]
 fixes_vcc_instructions: |
   Please put the SVN commit number in "commit" below, and any notes about how this
   was discovered in the "note" field.
 fixes:
    - commit: 941374ecdb99ccc316f6b527b8df0f1cf2bb80c0
      note: Use Jackson handler instrad of defualt for JSON
```
Comment (MattThyng) [inline cves/CVE-2017-15707.yml]:
```suggestion
     note: Use Jackson handler instead of default for JSON
```

File: cves/CVE-2017-15707.yml [hunk 2]
```diff
@@ -45,7 +46,11 @@ description_instructions: |
   that outsiders to Struts would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description:
+description: |
+  Struts was using an outdated library in order to read and process JSON files.
+  As a result, a special type of JSON file could be created that could cause a
+  DOS attack. Since the library was outdated and no longer being supported, this
```
Comment (Ksxerity) [inline cves/CVE-2017-15707.yml]: Consider writing out "Denial of Service" first before starting to use it as an acronym.

File: cves/CVE-2017-15707.yml [hunk 3]
```diff
@@ -205,4 +232,19 @@ mistakes:
...
+    discovered that a user could cause a DOS attack.
+
+    The team seems to have taken the proper initiatve of replacing the old
+    library with a new one (Jackson), that is still being worked on and
+    receiving security updates. It did take some time for them to switch things
+    over to using Jackson though as can be seen in some later commits showing
+    them switching to Jackson up to two months after the initial switch was
+    made. It was also noted that instead of fully removing the ability to use
+    the outdated json-lib library, they instead leave it as an optional thing,
+    with Jackson acting as the defualt JSON reader. They seem to making proper
```
Comment (Ksxerity) [inline cves/CVE-2017-15707.yml]: "They seem to be making"

File: cves/CVE-2017-15707.yml [hunk 4]
```diff
@@ -205,4 +232,19 @@ mistakes:
...
+    steps towards preventing the vulnerability, but still have some work to do,
+    as there was no mention of removing json-lib as optional.
```
Comment (Ksxerity) [inline cves/CVE-2017-15707.yml:250]: Maybe you can write out your thoughts on why they are keeping the json-lib as an option when it is clearly at risk of having multiple security vulnerabilities. Although Jackson was set as the default setting, I'm sure there are some users that may still opt to use json-lib without knowing fully why they switched over to Jackson.

**Comment by kleenip** [comment]
Going to give this +2 upvotes because it reminds me of the dependency fractal problems node.js has.

**Pull request merged**